File: simon-says.pl
#!/usr/bin/perl

use strict;
use warnings;

# Define the colors
my @colors = ("red", "green", "blue", "yellow");

# Define the game variables
my @sequence;
my $round = 1;
my $score = 0;

# Print the welcome message
print "Welcome to Simon Says!\n\n";

# Main game loop
while (1) {
    # Generate a new sequence for this round
    print "Round $round - Simon says...\n";
    push @sequence, $colors[int(rand(@colors))];
    print join(", ", @sequence) . "\n";

    # Get the user's input
    print "Repeat the sequence: ";
    chomp(my $input = <STDIN>);
    my @input_sequence = split(/,\s*/, $input);

    # Check the user's input
    my $correct = 1;
    for (my $i = 0; $i < @sequence; $i++) {
        if ($sequence[$i] ne $input_sequence[$i]) {
            $correct = 0;
            last;
        }
    }

    # Display the result
    if ($correct) {
        $score++;
        print "Correct! Your score is now $score.\n";
    } else {
        print "Incorrect! Game over. Your final score is $score.\n";
        last;
    }

    # Increase the round counter
    $round++;
}

File: simon-says.r
# Define a sequence of colors for Simon to say
simon_says <- sample(c("red", "blue", "green", "yellow"), 10, replace=TRUE)

# Print instructions
cat("Simon says: ", paste(simon_says, collapse=", "), "\n\n")

# Ask the player to repeat the sequence
cat("Repeat the sequence, separating each color with a comma: ")
player_input <- readline()

# Check if the player's input matches the sequence
if (player_input == paste(simon_says, collapse=",")) {
  cat("\nWell done! You repeated the sequence correctly.\n")
} else {
  cat("\nSorry, that's incorrect. The correct sequence was: ", paste(simon_says, collapse=", "), "\n")
}
